extensions/google-java/src/main/kotlin/com/connectrpc/extensions/GoogleJavaJSONAdapter.kt

```diff
@@ -16,54 +16,54 @@ package com.connectrpc.extensions
 
 import com.connectrpc.CODEC_NAME_JSON
 import com.connectrpc.Codec
-import com.google.protobuf.GeneratedMessageV3
 import com.google.protobuf.Internal
+import com.google.protobuf.Message
+import com.google.protobuf.TypeRegistry
 import com.google.protobuf.util.JsonFormat
 import okio.Buffer
 import okio.BufferedSource
 import okio.ByteString.Companion.encodeUtf8
 import kotlin.reflect.KClass
+import kotlin.reflect.cast
 
 /**
  * Adapter for Connect to use Google's protobuf-java runtime for
  * deserializing and serializing data types.
  */
-internal class GoogleJavaJSONAdapter<E : GeneratedMessageV3>(
```

Comment:
Two things: (1) we need to be able to weave a `TypeRegistry` through in order to use `google.protobuf.Any` with the JSON format. And (2) this type is way, too specific. In particular, this wouldn't work with dynamic messages, and there's no particular reason to require this type. So I instead opted for the more generic `Message`, since that is really all that is required.

Reply:
Aside: it's rather annoying that the JSON parser doesn't accept an `ExtensionRegistry`. I double-checked the implementation code and, sure enough, the JSON format in Java does not support parsing extension fields 😢.